Informal/Sets/Set-1/2/main.0.cpp

```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include <set>
+#include <vector>
+#include <sstream>
+#include <algorithm> 
+using namespace std;
+
+int refl(int index,int size){
+	int half = size%2?((size-1)/2):(size/2);
+	if(index<half) return size-1-index;
+	else return half-(index-half)-(!(size%2));
+}
+int charToPos(char c){
+	return c-'a';
+}
+
+void printVector(std::vector<char> v){
+	for(char a:v){
+		cout<<a;
+	}
+	cout<<"\n";
+}
+
+void solve(){
+	int size;
+	string s;
+	int exist[26] = {0};
+	int ind;
+	
+	cin>>size;
+	cin>>s;
+
+	vector<char>r;
+	vector<char>result;
+
+	//build 'exists' map, r string, O(n) 
```

Comment:
The final revision exists in this file, The idea is that i am first making a map between encoded and decoded letters, and then i am actually decoding the thing, so no std::find nessesary